spec/unit/provider/aws_internet_gateway_spec.rb

```diff
@@ -0,0 +1,150 @@
+require 'spec_helper'
+require 'chef_zero_rspec_helper'
+AWS.stub!
+
+describe Chef::Provider::AwsInternetGateway do
+  extend ChefZeroRspecHelper
+  let(:new_resource) do
+    Chef::Resource::AwsInternetGateway.new('my_igw', run_context)
+  end
+  let(:current_resource) do
+    Chef::Resource::AwsInternetGateway.new('my_igw', run_context)
+  end
+  let(:my_node) do
+    node = Chef::Node.new
+    node.automatic['platform'] = 'ubuntu'
+    node.automatic['platform_version'] = '12.04'
+    node
+  end
+  let(:events) { Chef::EventDispatch::Dispatcher.new }
+  let(:run_context) do
+    cookbook_collection = {}
+    Chef::RunContext.new(my_node, cookbook_collection, events)
+  end
+
+  let(:vpc_testme) { AWS::EC2::VPC.new('vpc-testme') }
+  let(:vpc_fakeme) { AWS::EC2::VPC.new('vpc-fakeme') }
+  let(:igw_testme) { AWS::EC2::InternetGateway.new('igw-testme') }
+
+  subject(:provider) do
```

Comment:
I don't like the `subject` method - I think a `let(:provider)` is clearer because then tests reference `provider` instead of `subject`.